### ISSUE TITLE
Avoid technical debt with custom types

### ISSUE THREAD

**Issue by Hyalunar** [Sleppo04/operationalspace#3]
Technical debt can build up if a limit set by a data type used in the past is now deprecated or needs to be changed.
Take for example the Unix 32bit-Timestamp vs the year 2038.

To avoid shortcomings on this end, I propose the introduction of type aliases using `typedef` to alias used types to a more descriptive name.

Example:
```C
uint16_t player_count
```
can be refactored to
```C
typedef uint16_t playercount_t
...


playercount_t players
```

**Comment by Sleppo04** [comment]
Agreed. Typedefs are free at runtime and provide cleaner and easier-to-maintain code.

Should the definitions be done in a central `definitions.h` or in the respective feature header file?

**Comment by Hyalunar** [comment]
I would like to do the definitions in the respective header files, because that's what you're going to include anyway if you plan on operating on such a field.
